flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/source/enumerator/splitter/MongoSplitVectorSplitter.java

```diff
@@ -0,0 +1,129 @@
+import com.mongodb.client.MongoClient;
+import org.apache.commons.collections.CollectionUtils;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonInt32;
+import org.bson.BsonValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.BSON_MAX_KEY;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.BSON_MIN_KEY;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.ERROR_MESSAGE_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.ID_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.ID_HINT;
+import static org.apache.flink.connector.mongodb.common.utils.MongoConstants.SPLIT_KEYS_FIELD;
+import static org.apache.flink.connector.mongodb.common.utils.MongoUtils.UNAUTHORIZED_ERROR;
+import static org.apache.flink.connector.mongodb.common.utils.MongoUtils.isCommandSucceed;
+import static org.apache.flink.connector.mongodb.common.utils.MongoUtils.splitVector;
+
+/**
+ * SplitVector Partitioner
+ *
+ * <p>Uses the SplitVector command to generate chunks for a collection. eg. <code>
+ * db.runCommand({splitVector:"inventory.products", keyPattern:{_id:1}, maxChunkSize:64})</code>
+ *
+ * <p>Requires splitVector privilege.
+ */
+@Internal
+public class MongoSplitVectorSplitter implements MongoSplitters.MongoSplitter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MongoSplitVectorSplitter.class);
+
+    public static final MongoSplitVectorSplitter INSTANCE = new MongoSplitVectorSplitter();
+
+    private MongoSplitVectorSplitter() {}
+
+    @Override
+    public Collection<MongoScanSourceSplit> split(MongoSplitContext splitContext) {
+        MongoClient mongoClient = splitContext.getMongoClient();
+        MongoNamespace namespace = splitContext.getMongoNamespace();
+        MongoReadOptions readOptions = splitContext.getReadOptions();
+
+        MemorySize chunkSize = readOptions.getPartitionSize();
+        // if partition size < 1mb, use 1 mb as chunk size.
+        int maxChunkSizeMB = Math.max(chunkSize.getMebiBytes(), 1);
+
+        BsonDocument keyPattern = new BsonDocument(ID_FIELD, new BsonInt32(1));
+
+        BsonDocument splitResult;
+        try {
+            splitResult = splitVector(mongoClient, namespace, keyPattern, maxChunkSizeMB);
+        } catch (MongoCommandException e) {
+            if (e.getErrorCode() == UNAUTHORIZED_ERROR) {
+                LOG.warn(
+                        "Unauthorized to execute splitVector command: {}, fallback to SampleSplitter",
+                        e.getErrorMessage());
+            } else {
+                LOG.warn(
+                        "Execute splitVector command failed: {}, fallback to SampleSplitter",
+                        e.getErrorMessage());
```

Comment:
We should fail hard in this case. The user told us to use splitVector, presumably for a good reason, so have them decide how to handle this issue.